File: Practica_3/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
    #Insertar elementos a la Cola
    def insert(self, e):
        if self.is_full():
            print("La cola esta llena!")
            return
        if self.is_empty():
            self.inicio = 0
            self.fin = 0
        else:
            self.fin = (self.fin + 1) % self.n
        self.arreglo[self.fin] = e
    #Eliminar elementos de la Cola
    def remove(self):
        if self.is_empty():
            print("La cola esta vacia!")
            return -1
        temp = self.arreglo[self.inicio]
        if self.inicio == self.fin:
            self.inicio = 0
            self.fin = -1
        else:
            self.inicio = (self.inicio + 1) % self.n
        return temp
    #
    def peek(self):
        if self.is_empty():
            print("La cola esta vacia!")
            return -1
        return self.arreglo[self.inicio]
    #Es Vacia
    def is_empty(self):
        return self.fin == -1
    #Es Llena
    def is_full(self):
        return (self.fin + 1) % self.n == self.inicio and self.fin != -1
    #Tamaño de la Cola
    def size(self):
        if self.is_empty():
            return 0
        if self.fin >= self.inicio:
            return self.fin - self.inicio + 1
        return self.n - self.inicio + self.fin + 1

#MAIN
class Main:
    print("\n<--COLAS-->")
    cola = Cola(5)
    
    # Insertar elementos
    cola.insert(1)
    cola.insert(2)
    cola.insert(3)
    cola.insert(4)
    cola.insert(5)
    
    # Mostrar el elemento en la parte frontal
    print("Elemento en el frente:", cola.peek())
    
    # Extraer elementos de la cola
    print("Elemento extraido:", cola.remove())
    print("Elemento extraido:", cola.remove())
    print("Elemento extraido:", cola.remove())
    print("Elemento extraido:", cola.remove())
    print("Elemento extraido:", cola.remove())
    
    # Verificar si la cola esta vacia
    print("¿La cola esta vacia?:", cola.is_empty())

File: Practica_3/src/Pila_Cola/Cola.java
package Pila_Cola;

public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;

    // Constructor
    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
    }

    // Insertar un elemento en la cola
    public void insert(long e) {
        if (isFull()) {
            System.out.println("La cola esta llena!");
            return;
        }
        if (isEmpty()) {
            inicio = 0;
            fin = 0;
        } else {
            fin = (fin + 1) % n;
        }
        arreglo[fin] = e;
    }

    // Eliminar un elemento de la cola
    public long remove() {
        if (isEmpty()) {
            System.out.println("La cola esta vacia!");
            return -1;
        }
        long temp = arreglo[inicio];
        if (inicio == fin) {
            inicio = 0;
            fin = -1;
        } else {
            inicio = (inicio + 1) % n;
        }
        return temp;
    }

    // Obtener elemento en la parte frontal
    public long peek() {
        if (isEmpty()) {
            System.out.println("La cola esta vacia!");
            return -1;
        }
        return arreglo[inicio];
    }

    // Verificar si la cola esta vacia
    public boolean isEmpty() {
        return fin == -1;
    }

    // Verificar si la cola esta llena
    public boolean isFull() {
        return (fin + 1) % n == inicio && fin != -1;
    }

    // Obtener el tamaño actual de la cola
    public int size() {
        if (isEmpty()) {
            return 0;
        }
        if (fin >= inicio) {
            return fin - inicio + 1;
        }
        return n - inicio + fin + 1;
    }
}
